src/prime_rl/trainer/sft/train.py

```diff
@@ -183,24 +183,22 @@ def train(config: SFTTrainerConfig):
                     dist.all_reduce(max_vio, op=dist.ReduceOp.MAX)
                     batch_max_vio += max_vio / grad_accum_steps
 
+                # Compute average loss over unmasked tokens
                 loss = loss[loss_mask].mean()
 
-                # Scale loss by number of gradient accumulation steps
-                loss /= grad_accum_steps
-
-                # Accumulate loss
-                batch_loss += loss
+                # Accumulate average loss over gradient accumulation steps
+                batch_loss += loss / grad_accum_steps
```

Comment:
```suggestion
                batch_loss += loss.detach() / grad_accum_steps
```
otherwise we will keep this in the graph but never backward on it